Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -0,0 +1,280 @@
+            const int k_probe = amrex::Math::floor((z_probe - prob_lo[2]) / cell_size[2]);
+
+            // get MultiFab data at lev
+            const amrex::MultiFab &Ex = warpx.getEfield(lev, 0);
+            const amrex::MultiFab &Ey = warpx.getEfield(lev, 1);
+            const amrex::MultiFab &Ez = warpx.getEfield(lev, 2);
+            const amrex::MultiFab &Bx = warpx.getBfield(lev, 0);
+            const amrex::MultiFab &By = warpx.getBfield(lev, 1);
+            const amrex::MultiFab &Bz = warpx.getBfield(lev, 2);
+
+            // Prepare interpolation of field components to cell center
+            // The arrays below store the index type (staggering) of each MultiFab, with the third
+            // component set to zero in the two-dimensional case.
+            amrex::IndexType const Extype = Ex.ixType();
+            amrex::IndexType const Eytype = Ey.ixType();
+            amrex::IndexType const Eztype = Ez.ixType();
+            amrex::IndexType const Bxtype = Bx.ixType();
+            amrex::IndexType const Bytype = By.ixType();
+            amrex::IndexType const Bztype = Bz.ixType();
+
+            int probe_proc = -1;
+            // MFIter loop to interpolate fields to cell center and get maximum values
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+            for (amrex::MFIter mfi(Ex, TilingIfNotGPU()); mfi.isValid(); ++mfi) {
+                const amrex::Box &box = amrex::enclosedCells(mfi.nodaltilebox());
+
+                if (box.contains(i_probe, j_probe, k_probe)) {
+                    // Make the box cell centered in preparation for the interpolation (and to avoid
+                    // including ghost cells in the calculation)
+                    const auto &arrEx = Ex[mfi].array();
+                    const auto &arrEy = Ey[mfi].array();
+                    const auto &arrEz = Ez[mfi].array();
+                    const auto &arrBx = Bx[mfi].array();
+                    const auto &arrBy = By[mfi].array();
+                    const auto &arrBz = Bz[mfi].array();
+
+                    amrex::ParticleReal Ex_interp = 0._rt, Ey_interp = 0._rt, Ez_interp = 0._rt;
+                    amrex::ParticleReal Bx_interp = 0._rt, By_interp = 0._rt, Bz_interp = 0._rt;
+
+                    amrex::Array<amrex::Real, 3> v_galilean = {{0}};
+                    const auto
+                        &xyzmin = WarpX::GetInstance().LowerCornerWithGalilean(box, v_galilean,
+                                                                               lev);
+                    const std::array<Real, 3> &dx = WarpX::CellSize(lev);
+                    amrex::GpuArray<amrex::Real, 3> dx_arr = {dx[0], dx[1], dx[2]};
+                    amrex::GpuArray<amrex::Real, 3> xyzmin_arr = {xyzmin[0], xyzmin[1], xyzmin[2]};
+
+                    //Interpolating to the probe point
+                    doGatherShapeN(x_probe, y_probe, z_probe, Ex_interp, Ey_interp, Ez_interp,
```

Comment:
In 2d, `y_probe` could be declared and set to 0. immediatly before this call. Alternately, it could be declared and set to 0. in the header file (and it would be unused after except here).